src/parser/ad_parser.js

```diff
@@ -237,47 +236,70 @@ function parseWrapper(wrapperElement) {
  */
 function parseExtensions(collection, extensions) {
   extensions.forEach(extNode => {
-    const ext = new AdExtension();
-    const extNodeAttrs = extNode.attributes;
-    const childNodes = extNode.childNodes;
+    const ext = parseExtension(extNode);
 
-    if (extNode.attributes) {
-      for (const extNodeAttrKey in extNodeAttrs) {
+    if (ext) {
+      collection.push(ext);
+    }
+  });
+}
+
+/**
+ * Parses an extension child node
+ * @param {Object} extNode - The extension node to parse
+ * @return {Object}
+ */
+function parseExtension(extNode) {
```

Comment:
extNode could have more explicit type of `Node`. And return should be `AdExtension|null`